addons/product/models/product_template.py

```diff
@@ -501,11 +501,6 @@ def create(self, vals_list):
 
     def write(self, vals):
         self._sanitize_vals(vals)
-        if 'uom_id' in vals or 'uom_po_id' in vals:
-            uom_id = self.env['uom.uom'].browse(vals.get('uom_id')) or self.uom_id
-            uom_po_id = self.env['uom.uom'].browse(vals.get('uom_po_id')) or self.uom_po_id
-            if uom_id and uom_po_id and uom_id.category_id != uom_po_id.category_id:
-                vals['uom_po_id'] = uom_id.id
```

Comment:
Note that there is a bug on these lines because it doesn't manage writing on multiple records at once correctly